src/main/java/org/wildfly/naming/client/util/NamingUtils.java

```diff
@@ -57,7 +57,7 @@ public static CompositeName toCompositeName(final Name name) throws InvalidNameE
             return (CompositeName) name;
         } else {
             final CompositeName compositeName = new CompositeName();
-            compositeName.add(name.toString());
+            compositeName.addAll(new CompositeName(name.toString()));
```

Comment:
There's still a case where a user explicitly creates a CompositeName which consists of (the string form of) a SimpleName plus some other name as well.  Such names are going to fail against an old server.

We may be able to assume, at least when talking to an old server, that the first component is always a SimpleName or other compound name that follows the syntax restrictions of the server's primary name space (i.e. left-to-right, delimited by slashes).  But beyond that we're on thin ice.  So it might be better to just ignore this case and tell users, if you're talking to an old server, you must either pass in a SimpleName, a decomposed CompositeName, or a string.